utils/yoke/src/kinda_sorta_dangling.rs

```diff
@@ -0,0 +1,83 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use core::mem::{ManuallyDrop, MaybeUninit};
+use core::ops::{Deref, DerefMut};
+
+/// This type is intended to be similar to the type `MaybeDangling<T>`
+/// proposed in [RFC 3336].
+///
+/// The effect of this is that in Rust's safety model, types inside here are not
+/// expected to have any memory dependent validity properties (`dereferenceable`, `noalias`).
+///
+/// See [#3696] for a testcase where `Yoke` fails this under miri's field-retagging mode.
+///
+/// This has `T: 'static` since we don't need anything
+/// else and we don't want to have to think (more) about variance.
+///
+/// After [RFC 3336] lands we can use `MaybeDangling` instead.
+///
+/// [RFC 3336]: https://github.com/rust-lang/rfcs/pull/3336
```

Comment:
Apologies for the self-plug, but for the sake of having a central implementation we can refer to, I do think it could be valuable to have a mention to https://docs.rs/maybe-dangling/0.1.0/maybe_dangling/struct.MaybeDangling.html as well 🙂 

_e.g._,

```rs
/// For those interested, note that this is an inlined version of the publicly available
/// <code>[::maybe_dangling::MaybeDangling][1]\<T></code> (but for the added
/// `T: 'static` bound).
///
/// [1]: https://docs.rs/maybe-dangling/^0.1.0/maybe_dangling/struct.MaybeDangling.html
```

Reply:
No, totally reasonable, will do.